src/chunker.c

```diff
@@ -8,13 +8,70 @@
  * optimization. See RFC1950 for details. */
 #define BYTES_PUSH_MAX 5552U
 
+/* Unless otherwise specified, take this as the default average chunk size */
+#define DEFAULT_CHUNK_SIZE_AVG (16 * 1024)
+
+static bool number_is_prime(unsigned int n) {
```

Comment:
unsigned int → size_t?

(after all, we only use this on size_t, and I think we should avoid type mismatches right away...)